Firebase/Auth/Source/FIRAuth.m

```diff
@@ -986,8 +988,8 @@ - (BOOL)canHandleNotification:(NSDictionary *)userInfo {
 }
 #endif
 
-- (BOOL)canHandleURL:(NSURL *)url {
-  return NO;
+- (BOOL)canHandleURL:(NSURL *)URL {
+  return [_authURLPresenter canHandleURL:URL];
```

Comment:
Use dispatch_sync for public method calling private method synchronously.

Reply:
done, thanks!